### PR TITLE
`4.17.0 Relevant` Deprecated NEO

### PR DESCRIPTION
**Pull request by CharlesDuboisSAP**
## What Has Changed?

See [Deprecated NEO #8180](https://github.wdf.sap.corp/MA/sdk/pull/8180#top)

## Manual Checks?

- [x] Check spelling and grammar, e.g., using Grammarly.
- [x] Verify links still work, e.g., if `id` or the name of a file is changed.
- [x] Update the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js), e.g., if a feature is added.

### CODE REVIEW

File: docs-java/faq.mdx [hunk 1]
```diff
@@ -311,8 +311,9 @@ Let us know if you miss functionality of the SAP BTP for the Cloud Foundry envir
 
 ### Do You Support SAP Business Technology Platform - Neo?
 
+We are planning to discontinue support of NEO with SAP Cloud SDK v5 by end of the year.
```
Comment (Johannes-Schneider) [inline docs-java/faq.mdx:314]: Who knows when somebody is going to stumble over this sentence ... 😉

```suggestion
We are planning to discontinue support of NEO with SAP Cloud SDK v5 by end of 2023.
```

Comment (Johannes-Schneider) [reply docs-java/faq.mdx:314]: Also, to make this stick out even more, why not wrap it inside a `:::warn` box and leave the rest of this paragraph untouched.

File: docs-java/faq.mdx [hunk 2]
```diff
@@ -311,8 +311,9 @@ Let us know if you miss functionality of the SAP BTP for the Cloud Foundry envir
...
+Therefore, we're announcing module deprecation in advance.
```
Comment (Johannes-Schneider) [inline docs-java/faq.mdx:315]: Personally, I don't think this sentence belongs here.

```suggestion
```

File: src/components/data/features.js [hunk 3]
```diff
@@ -252,9 +252,9 @@ export const features = [
     name: '[BTP Neo Environment](https://help.sap.com/viewer/ea72206b834e4ace9cd834feed6c0e09/Cloud/en-US)',
     category: 'Environment',
     java: {
-      status: YES,
+      status: DEPRECATED,
       docsLink: ``,
-      note: `Not recommended for new projects`
+      note: 'Deprecated since 4.17'
```
Comment (Johannes-Schneider) [inline src/components/data/features.js:257]: (minor)

```suggestion
      note: 'Deprecated since 4.17.0'
```

**Review by Johannes-Schneider — CHANGES_REQUESTED** [review]

File: docs-java/faq.mdx [hunk 4]
```diff
@@ -311,9 +311,11 @@ Let us know if you miss functionality of the SAP BTP for the Cloud Foundry envir
 
 ### Do You Support SAP Business Technology Platform - Neo?
 
-We still provide full support for SAP Business Technology Platform Neo.
-However, we do not recommend starting new projects with SAP BTP for the Neo environment.
+:::warning warning
+We are planning to discontinue support of NEO with SAP Cloud SDK v5 by end of 2023.
 The Cloud Foundry environment is better suited for apps and SAP S/4HANA extension development.
+:::
+We still provide full support for SAP Business Technology Platform Neo.
```
Comment (Johannes-Schneider) [inline docs-java/faq.mdx:318]: Let's add this back in:

```suggestion
:::

We still provide full support for SAP Business Technology Platform Neo.
However, we do not recommend starting new projects with SAP BTP for the Neo environment.
```

File: docs-java/faq.mdx [hunk 5]
```diff
@@ -311,9 +311,13 @@ Let us know if you miss functionality of the SAP BTP for the Cloud Foundry envir
...
 ### Do You Support SAP Business Technology Platform - Neo?
 
+:::warning warning
+We are planning to discontinue support of NEO with SAP Cloud SDK v5 by end of 2023.
+The Cloud Foundry environment is better suited for apps and SAP S/4HANA extension development.
+:::
+
 We still provide full support for SAP Business Technology Platform Neo.
 However, we do not recommend starting new projects with SAP BTP for the Neo environment.
-The Cloud Foundry environment is better suited for apps and SAP S/4HANA extension development.
```
Comment (Johannes-Schneider) [inline docs-java/faq.mdx:316]: Why are we keep removing sentences? 😅
Just add the box on top of this and we are good IMO.

**Review by Johannes-Schneider — APPROVED** [review]

File: docs-java/faq.mdx [hunk 6]
```diff
@@ -311,6 +311,10 @@ Let us know if you miss functionality of the SAP BTP for the Cloud Foundry envir
 
 ### Do You Support SAP Business Technology Platform - Neo?
 
+:::warning warning
+We are planning to discontinue support of NEO with SAP Cloud SDK v5 by end of 2023.
```
Comment (jjtang1985) [inline docs-java/faq.mdx:315]:
```suggestion
All the NEO specific classes and functions have been deprecated.
We are planning to discontinue support of NEO with SAP Cloud SDK v5 by end of 2023.
```

**Pull request merged**